### PR TITLE
_is_defined_in: Accept parent class locations too

### PR DESCRIPTION
**Pull request by sils1297**
CC @AbdealiJK for review

### CODE REVIEW

File: tests/test_bears/LinterBear.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    """
+    A simple bear to test that collectors are importing also bears that are
+    defined in another file *but* have baseclasses in the right file.
+
+    (linter will create a new class that inherits from this class.)
+    """
+
+    @staticmethod
+    def create_arguments(filename, file, config_file):
+        return ()
```
Comment (AbdealiJK) [inline tests/test_bears/LinterBear.py]: newline

File: coalib/collecting/Importers.py [hunk 2]
```diff
@@ -48,12 +48,19 @@ def _has_all(obj, attribute_names):
 
 
 def _is_defined_in(obj, file_path):
+    """
+    Chek if a class is defined in the given file.
```
Comment (AbdealiJK) [inline coalib/collecting/Importers.py]: Check *

Comment (sils1297) [inline coalib/collecting/Importers.py]: newcomer issue?

Comment (AbdealiJK) [inline coalib/collecting/Importers.py]: sure

Comment (sils1297) [inline coalib/collecting/Importers.py]: https://github.com/coala-analyzer/coala/issues/2058

**Comment by AbdealiJK** [comment]
ack 1b4c6a3

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2057/commits/dae744cb9b22042cc3c28da61d20aafc5983c19c

File: coalib/collecting/Importers.py [hunk 3]
```diff
@@ -47,15 +47,50 @@ def _has_all(obj, attribute_names):
...
-    except TypeError:  # Bool values and others
+    except TypeError:  # Builtin values don't have a source location
         pass
+
+    return False
+
+
+def _is_defined_in(obj, file_path):
+    """
+    Chek if a class is defined in the given file.
```
Comment (Makman2) [inline coalib/collecting/Importers.py]: typo :)

File: coalib/collecting/Importers.py [hunk 4]
```diff
@@ -47,15 +47,50 @@ def _has_all(obj, attribute_names):
...
+
+    Any class is considered to be defined in the given file if any of it's
+    parent classes or the class itself is defined in it.
+    """
+    if not inspect.isclass(obj):
+        return object_defined_in(obj, file_path)
+
+    for base in inspect.getmro(obj):
+        if object_defined_in(base, file_path):
+            return True
```
Comment (Makman2) [inline coalib/collecting/Importers.py]: can't you use `isinstance`?

File: coalib/collecting/Importers.py (hunk 3 above)
Comment (sils1297) [inline coalib/collecting/Importers.py]: we already filed a newcomer issue

File: coalib/collecting/Importers.py (hunk 4 above)
Comment (sils1297) [inline coalib/collecting/Importers.py]: not at all! It's a class, not an instance.

Comment (sils1297) [inline coalib/collecting/Importers.py]: or what do you mean?

**Comment by Makman2** [comment]
ack 08075d8

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**